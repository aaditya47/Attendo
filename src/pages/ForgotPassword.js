import { React, useState } from 'react';
import {
    Flex,
    Box,
    Heading,
    FormControl,
    FormLabel,
    Input,
    Link,
    Button,
    Grid,
    Image,
    useToast
} from '@chakra-ui/react';

import forgotPassword from '../assets/forgotPassword.svg'

export default function ForgotPassword() {
    const toast = useToast()
    const checkEmail = (email) => {
        var regexp = /\S+@\S+\.\S+/;
        return regexp.test(String(email).toLowerCase());
    }

    const onSubmit = (id, email) => {
        if (id && email) {
            let res = checkEmail(email)
            if (res === false) {
                toast({
                    position: 'top',
                    title: "Forgot Password Error",
                    description: "Enter a valid email ID",
                    status: "error",
                    duration: 9000,
                    isClosable: true,
                })
            }
        }
        else {
            toast({
                position: 'top',
                title: "Forgot Password Error",
                description: "Enter the complete credentials",
                status: "error",
                duration: 9000,
                isClosable: true,
            })
        }
    }
    const [id, setID] = useState('');
    const [email, setEmail] = useState('');
    return (
        <Grid templateColumns="repeat(2, 1fr)" gap={10}>
            <Flex width="full" align="center" justify="left">
                <Box p={8} borderWidth={1} borderRadius={8}>
                    <Box textAlign="center">
                        <Heading>
                            Forgot Password
                        </Heading>
                    </Box>
                    <Box my={4} textAlign="left">
                        <form>
                            <FormControl mt={6}>
                                <FormLabel>Roll Number</FormLabel>
                                <Input type="text" value={id} onChange={(event) => { setID(event.target.value) }} placeholder="Roll Number" />
                            </FormControl>
                            <FormControl mt={6}>
                                <FormLabel>Email ID</FormLabel>
                                <Input type="text" placeholder="Email ID" value={email} onChange={(event) => { setEmail(event.target.value) }} />
                            </FormControl>
                            <Button colorScheme="teal" variant="outline" width="full" mt={4} onClick={() => { onSubmit(id, email) }}>
                                <Link href="#">
                                    Get Password
                                </Link>
                            </Button>
                            <Button mt={4} colorScheme="teal" variant="link">
                                <Link href="/">
                                    Lost your Way? Click here
                                </Link>
                            </Button>
                        </form>
                    </Box>
                </Box>
            </Flex>
            <Box boxSize="sm" mt={10}>
                <Image src={forgotPassword} />
            </Box>
        </Grid>
    );
}

