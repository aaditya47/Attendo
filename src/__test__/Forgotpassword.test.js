import React from 'react';
import ReactDOM from 'react-dom';
import ForgotPassword from '../pages/ForgotPassword';
import { render, cleanup } from '@testing-library/react'

import renderer from 'react-test-renderer'

afterEach(cleanup)
it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<ForgotPassword></ForgotPassword>, div)
})
afterEach(cleanup)
it("renders button correctly", () => {
    const { getByTestId } = render(<ForgotPassword></ForgotPassword>)
    expect(getByTestId('button')).toHaveTextContent("Get Password")
});

it("matches snapshot", () => {
    const tree = renderer.create(<ForgotPassword></ForgotPassword>).toJSON();
    expect(tree).toMatchSnapshot();
})