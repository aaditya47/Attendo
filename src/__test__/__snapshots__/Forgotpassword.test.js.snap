// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="css-1pf1pew"
>
  <div
    className="css-12rrdgv"
  >
    <div
      className="css-11p92tb"
    >
      <div
        className="css-xi606m"
      >
        <h2
          className="chakra-heading css-0"
        >
          Forgot Password
        </h2>
      </div>
      <div
        className="css-jfzvvr"
      >
        <form>
          <div
            className="chakra-form-control css-19ug521"
            role="group"
          >
            <label
              className="chakra-form__label css-uvho5g"
              htmlFor="field-9622659948-5"
              id="field-9622659948-5-label"
            >
              Roll Number
            </label>
            <input
              className="chakra-input css-0"
              disabled={false}
              id="field-9622659948-5"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Roll Number"
              readOnly={false}
              required={false}
              type="text"
              value=""
            />
          </div>
          <div
            className="chakra-form-control css-19ug521"
            role="group"
          >
            <label
              className="chakra-form__label css-uvho5g"
              htmlFor="field-9622659948-6"
              id="field-9622659948-6-label"
            >
              Email ID
            </label>
            <input
              className="chakra-input css-0"
              disabled={false}
              id="field-9622659948-6"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Email ID"
              readOnly={false}
              required={false}
              type="text"
              value=""
            />
          </div>
          <button
            className="chakra-button css-ug24of"
            data-testid="button"
            onClick={[Function]}
            type="button"
          >
            Get Password
          </button>
          <button
            className="chakra-button css-q1ts0i"
            type="button"
          >
            <a
              className="chakra-link css-0"
              href="/"
            >
              Lost your Way? Click here
            </a>
          </button>
        </form>
      </div>
    </div>
  </div>
  <div
    className="css-1rgqx61"
  >
    <img
      className="chakra-image__placeholder css-0"
    />
  </div>
</div>
`;
